### PR TITLE
Update Python3 to 3.7.10, Backport CVE-2021-23336 patch to Python2

### PR DESCRIPTION
**Pull request by thcrain-msft**
NOTE: Change requires a possibly backwards incompatible change in a public-facing Python API. This requires a full test of all dependent packages before merging.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

CVE-2021-23336 is a high-severity web cache poisoning vulnerability in our python2 and python3 packages- let's fix that.

###### Change Log  

- Patch CVE-2021-23336 in Python2 using Gentoo backport
- Upgrade Python3 to 3.7.10 to fix CVE-2021-23336

###### Does this affect the toolchain?  

**YES**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-23336

###### Test Methodology

- Started a full pipeline build, starting with id 71582

### CODE REVIEW

File: SPECS/python2/python2.spec [hunk 1]
```diff
@@ -31,6 +31,8 @@ Patch10:        CVE-2007-4559.nopatch
 Patch11:        CVE-2019-18348.nopatch
 # CVE-2020-27619 patch backported from 3.6
 Patch12:        CVE-2020-27619.patch
+# CVE-2021-23336 patch backported from 3.6 courtesy of Gentoo
```
Comment (PawelWMS) [inline SPECS/python2/python2.spec:34]: Suggestion: also add a link to their backport for the sake of documenting our sources, even though we have the copy in the .patch file.

Comment (thcrain-msft) [reply SPECS/python2/python2.spec:34]: Added, thanks!

**Comment by thcrain-msft** [comment]
Test pipeline indicates that there were no regressions in package tests due to this change.

**Review by jslobodzian — APPROVED** [review]
:shipit:

**Pull request merged**